### ISSUE TITLE
Core Animation Engine regression: Can not handle top items next to a group

### ISSUE THREAD

**Issue by nevil** [airbnb/lottie-ios#1853]
## Which Version of Lottie are you using?

Lottie 4.0.1

## Expected Behavior 
When using the Main Thread Engine the initial (hollowed) star is displayed.

https://user-images.githubusercontent.com/36905/207494595-17bff305-6141-4d1a-9435-784650ee2955.mp4

## Actual Behavior
When using Core Animation Engine the initial star is not displayed.

https://user-images.githubusercontent.com/36905/207494643-7e0ae616-50cf-4da2-8d12-584c3227c3fa.mp4

## Animation JSON
[STAR_RATING_BUTTON_ANIMATION.json.zip](https://github.com/airbnb/lottie-ios/files/10223944/STAR_RATING_BUTTON_ANIMATION.json.zip)

## Discussion
The layer called "STAR_01" has top level items and a group.
The top level items next to "Group 1" can not be handled in `ShapeLayer.makeGroupLayers(...)`.
Additionally a fill (Fill 1) should be applied to the group ("Group 1").

Structure:
- Shape name: Path 2
- Fill name: Fill 2
- Group name: Group 1
   * Shape name: Path 1
   * ShapeTransform name: Transform
- Fill name: Fill 1

I attempted a solution by creating a group for the top level items, but I'm not sure how to handle the case of applying "Fill 1" to the correct group (Group 1).

```diff
    // If this shape doesn't have any groups but just has top-level shape items,
    // we can create a placeholder group with those items. (Otherwise the shape items
    // would be silently ignored, since we expect all shape layers to have a top-level group).
    if groupItems.isEmpty, parentGroup == nil {
      groupItems = [Group(items: otherItems, name: "")]
      otherItems = []
    }

+    if parentGroup == nil, !groupItems.isEmpty, !otherItems.isEmpty {
+        groupItems.append(Group(items: otherItems, name: "Other Items"))
+        otherItems = []
+    }

    // `ShapeItem`s either draw a path, or modify how a path is rendered.
```

**Comment by calda** [comment]
Thanks for the report and animation json -- I'll find a chance to look in to this soon

**Comment by calda** [comment]
Fixed in #1874, thanks for the report!

**Comment by nevil** [comment]
@calda 

Thanks for fixing so fast.
I confirmed on my side as well and the issue is fixed 🎉
